bridges/snowbridge/primitives/inbound-queue/src/v2/converter.rs

```diff
@@ -89,19 +102,64 @@ where
 		fn prepare(message: Message) -> Result<PreparedMessage, ConvertMessageError> {
 			let mut remote_xcm: Xcm<()> = Xcm::new();
 
-			// Allow xcm decode failure so that assets can be trapped on AH instead of this
-			// message failing but funds are already locked on Ethereum.
-			if let Ok(versioned_xcm) = VersionedXcm::<()>::decode_with_depth_limit(
-				MAX_XCM_DECODE_DEPTH,
-				&mut message.xcm.as_ref(),
-			) {
-				if let Ok(decoded_xcm) = versioned_xcm.try_into() {
-					remote_xcm = decoded_xcm;
-				}
-			}
-
 			let ether_location = Location::new(2, [GlobalConsensus(EthereumNetwork::get())]);
 
+			match &message.xcm {
+				XcmCommand::Raw(raw) => {
+					// Allow xcm decode failure so that assets can be trapped on AH instead of this
+					// message failing but funds are already locked on Ethereum.
+					if let Ok(versioned_xcm) = VersionedXcm::<()>::decode_with_depth_limit(
+						MAX_XCM_DECODE_DEPTH,
+						&mut raw.as_ref(),
+					) {
+						if let Ok(decoded_xcm) = versioned_xcm.try_into() {
+							remote_xcm = decoded_xcm;
+						}
+					}
+				},
+				XcmCommand::TokenRegistration { token, network: _ } => {
+					let chain_id = match EthereumNetwork::get() {
+						Ethereum { chain_id } => chain_id,
+						_ => panic!("checked statically, qed"),
+					};
+					let bridge_owner =
+						EthereumLocationsConverterFor::<[u8; 32]>::from_chain_id(&chain_id);
+					let dot_asset = Location::new(1, Here);
+					let dot_fee: xcm::prelude::Asset = (dot_asset, CreateAssetDeposit::get()).into();
+
+					let eth_asset_value = 9_000_000_000_000u128; // TODO needs to be specified
```

Comment:
@vgeddes we need to specify the asset deposit value in Ether. The DOT value of the asset deposit fee can be configured, but we also need to provide an ether value for the `ExchangeAsset` command. Would it make sense if I add it here? https://github.com/Snowfork/snowbridge/pull/1381/files#diff-bf4d91d5f66dce91d0ec3233509f8fc602d4abad3f52142c907be67e784eeb7cR59

Reply:
Hey, yeah, definitely. I assigned https://github.com/Snowfork/snowbridge/pull/1381 to you so you can get more familiar with Solidity.

Reply:
Is `message.value` not sufficient for this? It is the amount of ether that the user provided.